public/app/features/dashboard-scene/scene/LibraryPanelBehavior.tsx

```diff
@@ -61,8 +55,20 @@ export class LibraryPanelBehavior extends SceneObjectBase<LibraryPanelBehaviorSt
 
     this.setState({ _loadedPanel: libPanel, isLoaded: true, name: libPanel.name, title: libPanelModel.title });
 
-    const clone = vizPanel.clone(vizPanelState);
-    gridItem.setState({ body: clone });
+    vizPanel.setState(vizPanelState);
+    vizPanel.changePluginType(libPanelModel.type, vizPanelState.options, vizPanelState.fieldConfig);
+
+    const layoutElement = vizPanel.parent!;
+
+    // Migrate repeat options to layout element
+    if (libPanelModel.repeat && layoutElement instanceof DashboardGridItem) {
+      layoutElement.setState({
+        variableName: libPanelModel.repeat,
+        repeatDirection: libPanelModel.repeatDirection === 'h' ? 'h' : 'v',
+        maxPerRow: libPanelModel.maxPerRow,
+      });
+      layoutElement.performRepeat();
+    }
```

Comment:
I think moving this here creates some issues, not sure why though: creating a new dashboard, then adding a variable and a repeated lib panel and then saving that dashboard does not render them after it auto-exits edit mode. Also won't render on reload. The panels are there as I can briefly see the "Loading panels" message, and they are also saved on the dashboard json, but they won't properly render. 

This is really weird cuz at some point after fiddling with the dashboard it does start working and further page reloads also start working... It's very weird.

Reply:
I pushed a commit with the fix for this, the gridItem was lacking height 😅 , now it renders propperly